projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts

```diff
@@ -1,21 +1,125 @@
 import { expect, test } from '@jupyterlab/galata';
+import { assert } from 'console';
 
 /**
  * Don't load JupyterLab webpage before running the tests.
  * This is required to ensure we capture all log messages.
  */
 test.use({ autoGoto: false });
+const delay = ms => new Promise(res => setTimeout(res, ms));
```

Comment:
nit: so the tests are grouped logically by some criteria,
for example user input validation tests.
Then we have happy path coverage, like download job then verify job had been downloaded the way expected

Reply:
Further tests should be added. I will push these ones for now,  and the others can be discussed latter while adding other tests for the functionalities. Current tests mainly focus on the JSX components that are being created.